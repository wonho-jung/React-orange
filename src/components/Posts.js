import React, { useState } from 'react';
import Post from './Post';
import styled from 'styled-components';

function Posts({ data }) {
  // get newdata array from post components filter function
  const [newdata, setNewdata] = useState(data);

  return (
    <PostsContainer>
      {data &&
        newdata.map((item) => (
          <Post
            data={newdata}
            setNewdata={setNewdata}
            key={item.id}
            title={item.title}
            body={item.body}
            author={item.author.name}
            publishData={item.publishedAt}
          />
        ))}
      {!data && <h1>Sorry Refresh your page :(</h1>}
    </PostsContainer>
  );
}

export default Posts;

const PostsContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
`;
