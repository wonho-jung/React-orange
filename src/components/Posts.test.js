import React from 'react';
import renderer from 'react-test-renderer';
import Posts from './Posts';

describe('Posts component', () => {
  it('matchs the snapshot', () => {
    const tree = renderer.create(<Posts />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
