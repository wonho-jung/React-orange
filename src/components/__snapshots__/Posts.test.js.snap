// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component matchs the snapshot 1`] = `
<div
  className="sc-gtsrHT iUuxbj"
>
  <h1>
    Sorry Refresh your page :(
  </h1>
</div>
`;
